File: hardware_interface/camera/Kinect.cpp
#include <opencv2/core/core.hpp>
#include <opencv2/highgui/highgui.hpp>
#include <opencv2/imgproc/imgproc.hpp>
#include <stdio.h>

using namespace cv;

int main(int argc, char **argv) {
  Mat image;
  if (argc < 2) {
    printf("Please specify a device\n");
    return 1;
  }
  VideoCapture cam(argv[1]);
  if (!cam.isOpened()) {
    printf("Cannot open device\n");
    return 1;
  }
  namedWindow("display", CV_WINDOW_AUTOSIZE);
  while (true) {
    cam.read(image);
    imshow("display", image);
    waitKey(0);
  }
  cam.release();
  return 0;
}

File: hardware_interface/camera/Kinect.py
import numpy as np
import cv2 # computer vision
import sys

if len(sys.argv) < 2:
  print "not enough args"
  sys.exit(0)

cap = cv2.VideoCapture(int(sys.argv[1]))

if not cap.isOpened():
  print "cant open device"
  sys.exit(0)

while(True):
  try:
    print "reading frame"
    ret, frame = cap.read()
    print "Got frame"
    cv2.imshow('frame', frame)
    if cv2.waitKey(1) == ord('q'):
      break
  except:
    print "cannot read frame"

cap.release()
cv2.destroyAllWindows()
